rasa/shared/core/events.py

```diff
@@ -377,27 +409,31 @@ def _from_parse_data(
         )
 
     def __hash__(self) -> int:
+        """Returns unique hash of object."""
         return hash((self.text, self.intent_name, jsonpickle.encode(self.entities)))
```

Comment:
I don't think `self.entities` here is correct, since it contains a lot of irrelevant unnecessary info

Reply:
fixed it by using `self.substate` for hashing